src/main/java/seedu/address/logic/parser/ListEventsCommandParser.java

```diff
@@ -2,38 +2,37 @@
 
 import static seedu.address.logic.parser.CliSyntax.PREFIX_SORT;
 
-import seedu.address.logic.commands.ListCommand;
+import seedu.address.logic.commands.ListEventsCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
-import seedu.address.model.person.SortField;
-
+import seedu.address.model.event.EventSortField;
 
 /**
- * Parses input arguments and creates a new ListCommand object
+ * Parses input arguments and creates a new ListEventsCommand object.
  */
-public class ListCommandParser implements Parser<ListCommand> {
+public class ListEventsCommandParser implements Parser<ListEventsCommand> {
 
     /**
-     * Parses the given {@code String} of arguments in the context of the ListCommand
-     * and returns a {@code ListCommand} for execution.
+     * Parses the given {@code String} of arguments in the context of the ListEventsCommand
+     * and returns a {@code ListEventsCommand} for execution.
      *
      * @throws ParseException if the user input does not conform to the expected format
      */
-    public ListCommand parse(String args) throws ParseException {
+    public ListEventsCommand parse(String args) throws ParseException {
 
-        SortField sortField;
+        EventSortField sortField;
         ArgumentMultimap argMultiMap = ArgumentTokenizer.tokenize(args, PREFIX_SORT);
 
```

Comment:
This few lines of code would help catch the case when random words are added behind the command, e.g listEvents randomwords s/E and throw an exception. Currently, it'll continue to return a sorted events field. These few lines of code will be helpful for listPersons as well. (Thou I haven't tested it yet)
```suggestion
        if (!argMultiMap.getValue(PREFIX_SORT).isPresent()
                || !argMultimap.getPreamble().isEmpty()) {
            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListEventsCommand.MESSAGE_USAGE));
        }

```

Reply:
Hmm yea Qiaoran also raised [this issue](https://github.com/AY2223S1-CS2103-F09-2/tp/pull/53#discussion_r992221499) when I did sort persons. I decided then that not being strict about input validation was a good thing. But seeing how other commands like `add` also disallow extra input before the prefixes, I think I will change the logic to enforce strict user input.

Reply:
Actually this check for the IF statement `!argMultiMap.getValue(PREFIX_SORT).isPresent()` will mean `listEvents` is an invalid command!

I believe the solution is to only check if the preamble is not empty. This way `listEvents` and `listEvents s/e` will work, while `listEvents rubbish s/e` will throw an exception.

An input like `listEvents s/n rubbish` will be interpreted as an invalid sort field (`n rubbish`) in `ParserUtil#parseEventSortField()` and an exception will also be thrown.

Reply:
Oh yes, you're right!